plugin-lifecycle/src/main/java/com/mapbox/maps/plugin/lifecycle/MapboxLifecyclePluginImpl.kt

```diff
@@ -33,14 +33,23 @@ class MapboxLifecyclePluginImpl : MapboxLifecyclePlugin {
           you need manually invoke the corresponding lifecycle methods in onStart/onStop/onDestroy/onLowMemory methods of the host Activity"""
       )
     } else {
-      val componentCallback = object : ComponentCallbacks {
+      val componentCallback = object : ComponentCallbacks2 {
         override fun onConfigurationChanged(newConfig: Configuration) {
           // no need
         }
 
         override fun onLowMemory() {
           observer.onLowMemory()
         }
+
+        override fun onTrimMemory(level: Int) {
+          when (level) {
+            ComponentCallbacks2.TRIM_MEMORY_RUNNING_CRITICAL, ComponentCallbacks2.TRIM_MEMORY_RUNNING_LOW -> {
+              observer.onLowMemory()
```

Comment:
imho we should also log here, it may help to trace some customer issues making sure we indeed call `onLowMemory()`

Reply:
agreed, added warning log when we actually call `observer.onLowMemory()`